src/librustdoc/html/static/main.js

```diff
@@ -868,19 +863,22 @@ function defocusSearchBar() {
             // Check for type name and type generics (if any).
             function checkType(obj, val, literalSearch) {
                 var lev_distance = MAX_LEV_DISTANCE + 1;
-                var x;
+                var len;
                 if (obj[NAME] === val.name) {
                     if (literalSearch === true) {
                         if (val.generics && val.generics.length !== 0) {
                             if (obj.length > GENERICS_DATA &&
                                   obj[GENERICS_DATA].length >= val.generics.length) {
                                 var elems = obj[GENERICS_DATA].slice(0);
                                 var allFound = true;
+                                var x, elen, firstGeneric;
 
-                                for (var y = 0; allFound === true && y < val.generics.length; ++y) {
+                                len = val.generics.length;
+                                for (var y = 0; allFound === true && y < len; ++y) {
                                     allFound = false;
-                                    var firstGeneric = getObjectFromId(val.generics[y]).name;
-                                    for (x = 0; allFound === false && x < elems.length; ++x) {
+                                    firstGeneric = getObjectFromId(val.generics[y]).name;
+                                    elen = elems.length;
+                                    for (x = 0; allFound === false && x < elen; ++x) {
                                         allFound = getObjectFromId(elems[x]).name === firstGeneric;
                                     }
                                     if (allFound === true) {
```

Comment:
```suggestion
var allFound = elems.every(function(id) { return getObjectFromId(id).name == firstGeneric; });
if (allFound === true) {
  elems.pop();
}
```
If you don't like the suggestion, you have a typo: `elen` -> `elems`.

Reply:
It's not a typo, it's meant for "elements length". But it's not that great so I'll update it a bit. Also, the code you suggested might be a worth a try, but I'll delay it for another PR until I fully get how it works.